sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteWithExpression.scala

```diff
@@ -68,9 +68,15 @@ object RewriteWithExpression extends Rule[LogicalPlan] {
 
   private def applyInternal(p: LogicalPlan): LogicalPlan = {
     val inputPlans = p.children
+    val commonExprIdSet = p.expressions
+      .flatMap(_.collect { case r: CommonExpressionRef => r.id })
```

Comment:
We should avoid collecting references from nested With, as this rule skips nested With and leaves it to the next iteration.

Reply:
I understand that this set is only used to determine whether it is inline and does not affect the logic of nest with. Can you give a specific example?